cudax/include/cuda/experimental/__event/event.cuh

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX_EVENT_DETAIL_H
+#define _CUDAX_EVENT_DETAIL_H
+
+/*
+    event synopsis
+namespace cuda::experimental {
+class event : public event_ref {
+public:
+    enum class flags : unsigned int { none, blocking_sync, interprocess };
+
+    event(stream_ref, flags = flags::none);
+    event(uninit_t) noexcept;
+    event(event&&) noexcept;
+    ~event();
+    event& operator=(event&&) noexcept;
+
+    [[nodiscard]] static event from_native_handle(cudaEvent_t) noexcept;
+    static event from_native_handle(int) = delete;
+    static event from_native_handle(nullptr_t) = delete;
+
+    [[nodiscard]] cudaEvent_t release() noexcept;
+
+    [[nodiscard]] friend flags operator|(flags, flags) noexcept;
+
+    // From event_ref:
+    using value_type = cudaEvent_t;
+
+    void record(stream_ref) const;
+
+    void wait(stream_ref) const;
+
+    [[nodiscard]] cudaEvent_t get() const noexcept;
+
+    [[nodiscard]] explicit operator bool() const noexcept;
+
+    [[nodiscard]] friend bool operator==(event_ref, event_ref);
+    [[nodiscard]] friend bool operator!=(event_ref, event_ref);
+};
+}  // cuda::experimenal
+*/
```

Comment:
I am not really a fan of the synopsis comments. They are most often outdated and we actually have the doxygen documentation so it feels like a waste of space